veps/sig-storage/incremental-backup.md

```diff
@@ -367,6 +404,232 @@ Disadvantages:
+
+Ingress to the virt-launcher Pod is currently impossible because all incoming traffic is redirected to the guest VM. To address this limitation, several alternative approaches have been considered.
+
+### Chosen M.O. - Export Type: Backup
+
+KubeVirt currently supports a `VirtualMachineExport` API which facilitates the secure export of volume data over HTTPS. This is achieved by the export controller reconciling the `VirtualMachineExport` CR to spin up a temporary virt-exportserver pod. This pod mounts the source volumes and exposes them via an internal ClusterIP service and, optionally, an external Ingress/Route through the export proxy deployment.
+
+This proposal generalizes the `VirtualMachineExport` implementation to support a new source kind: `VirtualMachineBackup`.
+
+The flow is as follows:
+1. A user creates a `VirtualMachineBackup` CR with spec.mode: Pull.
+2. The backup controller invokes the `Backup` RPC with `Start` cmd on the target virt-launcher. This triggers the creation of the associated libvirt backup job and spins up the NBD server inside the virt-launcher pod, listening on a Unix domain socket (e.g., `/run/kubevirt/sockets/backup-nbd.sock`).
+3. Once the backup job is progressing, the backup controller automatically creates an owned `VirtualMachineExport` CR, setting its source to the `VirtualMachineBackup`.
+4. The export-controller reconciles the new `VirtualMachineExport`. It identifies the source type and launches the virt-exportserver pod.
+  - Internal Listener: The server is configured to listen on a protected internal port (9090) for the backup tunnel.
+  - External Listener: The server listens on the standard external port (8443) for user HTTP traffic.
+  - Configuration: The controller injects the included volume list (from the `VirtualMachineBackupStatus`) into the pod's environment variables.
+5. Once the export service is ready, the backup controller retrieves the service's name and the internal CA certificate. It then issues yet another `Backup` RPC, this time with the `Connect` cmd to the virt-launcher.
+6. The virt-launcher acts as the TLS client. It establishes a persistent, self-healing TLS tunnel to the virt-exportserver's internal port. This reverse-tunnel architecture ensures no network changes are required for the virt-launcher pod.
+7. Data Serving:
+  - The virt-exportserver registers dynamic HTTP handlers for every volume included in the backup under the new `Backups` status field.
+  - The backup controller recognizes that the `VirtualMachineExport` is serving and updates the `Ready` condition of the `VirtualMachineBackupStatus` to true.
+  - The `VirtualMachineBackupStatus` includes the created URLs for every volume in its `includedVolumes`'s list by populating the `mapEndpoint` and `dataEndpoint` fields respectively.
+  - When a client requests a specific endpoint, the server translates this into a JSON-RPC command (`read`, `map`, or `info`) and sends it over the TLS tunnel. 
```

Comment:
Why JSON RPC and not REST? I'm not necessarily opposed to JSON RPC. May make things a slightly more complicated for clients using curl. I don't think we use it anywhere else in kubevirt.

Maybe if we have virtctl wrappers for client API it is less of a concern. Have you thought about that (regardless of API implementation)?